### PR TITLE
[GLUTEN-7795][CH] Add backend task id log

### PR DESCRIPTION
**Pull request by loneylee**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#7795)

## How was this patch tested?

Test by local log

### CODE REVIEW

**Comment by loneylee** [comment]
Run Gluten Clickhouse CI

**Comment by loneylee** [comment]
Run Gluten Clickhouse CI

**Review by liuneng1994 — APPROVED** [review]
LGTM

File: backends-clickhouse/src/main/java/org/apache/gluten/memory/CHThreadGroup.java [hunk 1]
```diff
@@ -28,7 +28,13 @@ public class CHThreadGroup implements TaskResource {
    */
   public static void registerNewThreadGroup() {
     if (TaskResources.isResourceRegistered(CHThreadGroup.class.getName())) return;
-    CHThreadGroup group = new CHThreadGroup();
+
+    String taskId = "";
```
Comment (liuneng1994) [inline backends-clickhouse/src/main/java/org/apache/gluten/memory/CHThreadGroup.java:32]:
```suggestion
    String taskId;
```

File: backends-clickhouse/src/main/java/org/apache/gluten/memory/CHThreadGroup.java [hunk 2]
```diff
@@ -28,7 +28,13 @@ public class CHThreadGroup implements TaskResource {
...
+    if (TaskResources.getLocalTaskContext() != null) {
```
Comment (liuneng1994) [inline backends-clickhouse/src/main/java/org/apache/gluten/memory/CHThreadGroup.java:33]:
```suggestion
    if (!TaskResources.getLocalTaskContext()) {
```

File: cpp-ch/local-engine/Common/QueryContext.cpp [hunk 3]
```diff
@@ -124,14 +126,24 @@ std::shared_ptr<DB::ThreadGroup> QueryContext::currentThreadGroup()
     throw Exception(ErrorCodes::LOGICAL_ERROR, "Thread group not found.");
 }
 
-void QueryContext::logCurrentPerformanceCounters(ProfileEvents::Counters & counters) const
+String QueryContext::currentTaskIdOrEmpty()
+{
+    if (auto thread_group = CurrentThread::getGroup())
+    {
+        const int64_t id = reinterpret_cast<int64_t>(CurrentThread::getGroup().get());
```
Comment (liuneng1994) [inline cpp-ch/local-engine/Common/QueryContext.cpp:133]:
```suggestion
        const int64_t id = reinterpret_cast<int64_t>(thread_group.get());
```

File: cpp-ch/local-engine/Common/QueryContext.cpp [hunk 4]
```diff
@@ -83,19 +84,20 @@ DB::ContextPtr QueryContext::globalContext()
     return Data::global_context;
 }
 
-int64_t QueryContext::initializeQuery()
+int64_t QueryContext::initializeQuery(String task_id)
```
Comment (liuneng1994) [inline cpp-ch/local-engine/Common/QueryContext.cpp:87]:
```suggestion
int64_t QueryContext::initializeQuery(const String & task_id)
```

File: cpp-ch/local-engine/Common/QueryContext.h [hunk 5]
```diff
@@ -40,10 +40,11 @@ class QueryContext
         static QueryContext instance;
         return instance;
     }
-    int64_t initializeQuery();
+    int64_t initializeQuery(String task_id);
```
Comment (liuneng1994) [inline cpp-ch/local-engine/Common/QueryContext.h:43]:
```suggestion
    int64_t initializeQuery(const String& task_id);
```

File: cpp-ch/local-engine/Common/QueryContext.h [hunk 6]
```diff
@@ -40,10 +40,11 @@ class QueryContext
...
     DB::ContextMutablePtr currentQueryContext();
+    String currentTaskIdOrEmpty();
     static std::shared_ptr<DB::ThreadGroup> currentThreadGroup();
-    void logCurrentPerformanceCounters(ProfileEvents::Counters & counters) const;
+    void logCurrentPerformanceCounters(ProfileEvents::Counters & counters, String task_id) const;
```
Comment (liuneng1994) [inline cpp-ch/local-engine/Common/QueryContext.h:47]:
```suggestion
    void logCurrentPerformanceCounters(ProfileEvents::Counters & counters, const String& task_id) const;
```

File: cpp-ch/local-engine/Common/QueryContext.cpp [hunk 7]
```diff
@@ -124,14 +126,24 @@ std::shared_ptr<DB::ThreadGroup> QueryContext::currentThreadGroup()
...
+{
+    if (auto thread_group = CurrentThread::getGroup())
+    {
+        const int64_t id = reinterpret_cast<int64_t>(CurrentThread::getGroup().get());
+        return query_map_.get(id)->task_id;
+    }
+   return "";
+}
+
+void QueryContext::logCurrentPerformanceCounters(ProfileEvents::Counters & counters, String task_id) const
```
Comment (liuneng1994) [inline cpp-ch/local-engine/Common/QueryContext.cpp:139]:
```suggestion
void QueryContext::logCurrentPerformanceCounters(ProfileEvents::Counters & counters, const String& task_id) const
```

File: cpp-ch/local-engine/Common/QueryContext.cpp [hunk 8]
```diff
@@ -124,14 +126,24 @@ std::shared_ptr<DB::ThreadGroup> QueryContext::currentThreadGroup()
     throw Exception(ErrorCodes::LOGICAL_ERROR, "Thread group not found.");
 }
 
-void QueryContext::logCurrentPerformanceCounters(ProfileEvents::Counters & counters) const
+String QueryContext::currentTaskIdOrEmpty()
```
Comment (liuneng1994) [inline cpp-ch/local-engine/Common/QueryContext.cpp:129]:
```suggestion
String QueryContext::currentTaskIdOrEmpty() const
```

**Comment by loneylee** [comment]
Run Gluten Clickhouse CI on x86

**Pull request merged**